### DISCUSSION TITLE
The cypress npm package is installed, but the Cypress binary is missing.

### DISCUSSION THREAD

**Discussion by Muhammadehsan2019** [cypress-io/cypress#15939] [CI setup]
@bahmutov 
@jennifer-shehane 
I am trying to integrate Cypress with Jenkins using a FreeStyle Project.  I am getting the below error message: 
I have followed the directions on https://github.com/cypress-io/cypress/issues/2634
but when I use CYPRESS_CACHE_FOLDER=cache/Cypress or CYPRESS_CACHE_FOLDER=./node_modules/cypress/cache/Cypress npx cypress open I get the error that  CYPRESS_CACHE_FOLDER is not recognized. 

**The cypress npm package is installed, but the Cypress binary is missing.**

We expected the binary to be installed here: C:\Windows\system32\config\systemprofile\AppData\Local\Cypress\Cache\6.8.0\Cypress\Cypress.exe
 
Reasons it may be missing:
 
- You're caching 'node_modules' but are not caching this path: C:\Windows\system32\config\systemprofile\AppData\Local\Cypress\Cache
- You ran 'npm install' at an earlier build step but did not persist: C:\Windows\system32\config\systemprofile\AppData\Local\Cypress\Cache
 
Properly caching the binary will fix this error and avoid downloading and unzipping Cypress.
 
Alternatively, you can run 'cypress install' to download the binary again.
 
https://on.cypress.io/not-installed-ci-error

**Comment by bahmutov** [comment]
Can you provide more details like the full CI config file? Do you set the `CYPRESS_CACHE_FOLDER` for all steps? Are you caching the `CYPRESS_CACHE_FOLDER` folder and restoring it before the next install / run?

**Reply by Muhammadehsan2019** [reply]
@bahmutov 
@jennifer-shehane 
I don't know how to cache the CYPRESS_CACHE_FOLDER on JENKINS.
I used CYPRESS_CACHE_FOLDER=npm install, but it throws an error that CYPRESS_CACHE_FOLDER is not recognized.
below are the snapshots for the CI configuration

**Comment by Muhammadehsan2019** [comment]
@bahmutov 
Could you please take a look at the CI configuration. This is how it looks for me.

**Reply by bahmutov** [reply]
I do not see any commands to cache the binary unfortunately in your screenshots. Please consult the Jenkins information on how to cache folders with dependencies. Our documentation for caching Cypress is at https://on.cypress.io/caching

**Reply by Muhammadehsan2019** [reply]
@bahmutov 
I already went through that, but that doesn't give any live example of how to cache folders with dependencies. I have so many problems dealing with Cypress and yet no one ever helped me. I don't know where to get help from? you are recommending something that I already went through but doesn't help at all. at least you can comment on the pictures I uploaded and tell me where to cache the folder with dependencies.

**Reply by bahmutov** [reply]
I am sorry to say, but we do not control folder caching on your CI, it is up to you to configure it. We give a lot of examples of caching in our https://github.com/cypress-io/cypress-example-kitchensink for CI services, but unfortunately there is not hosted Jenkins service for us to use, thus we do not have an example for Jenkins. I could also suggest looking at our "generic CI" steps in https://github.com/cypress-io/cypress-workshop-ci/tree/main/slides/generic-ci#caching This is not Jenkins, but shows the general idea of caching.

Tip: after installing Cypress you can see where it caches its binary by calling `cypress info` or `cypress cache path` (if you have installed Cypress globally). If you are installing cypress locally use `npx cypress info` or `npx cypress cache path`
